core/services/workflows/v2/engine.go

```diff
@@ -234,8 +254,64 @@ func (e *Engine) handleAllTriggerEvents(ctx context.Context) {
 	}
 }
 
-func (e *Engine) startNewWorkflowExecution(_ context.Context, _ capabilities.TriggerResponse) {
-	// TODO(CAPPL-735): implement execution phase
+// new workflow execution, blocking until completed
```

Comment:
nit: I know it's not a public function, but I would still do startExecution as the first word, like public documentation expects in Go. It makes refactoring it to public easier (not that I foresee that for this function, but I've made many private functions and exposed them later) and keeps things consistent.